word_guess.rb

```diff
@@ -0,0 +1,325 @@
+  def play_game
+    if win?(@guessed_things)
+      exit
+    end
+
+    if lose?
+      exit
+    end
+
+    print_answer_blanks
+    display_guessed_letters
+    ask_user_for_guess
+
+    if guessed_word?
+      if win?(@last_guess_array)
+        exit
+      end
+    end
+
+      add_to_guess_array
+      update_remaining_guesses
+
+  end
+
+  def win?(array) #my collarborator was a match teacher and she undertands intersections!
+    if @answer_validation_array & array ==  @answer_validation_array.uniq # must use uniq beacause intersection won't return duplicate letters. Otherwise can never win the game!(with words that have duplicate letters)
+      puts "You managed to keep the dinosaur at bay and correctly guessed the word, '#{ @answer_validation_array.join("") }'." #not-match teacherese: when the guessed letter array includes all the letters in the answer array, the "intersection" will be the answer array.
+      puts "You win!"
+      return true
+    else
+      return false
+    end
+  end
+
+  def lose?
+    if @number_of_guesses == 0
+      puts "The dinosaur chomps you before you are able to guess the word, '#{ @answer_validation_array.join("") }'."
+      puts "You were eaten :("
+      return true
+    else
+      return false
+    end
+  end
+
+  def print_answer_blanks
+    puts "==================================================================\n"
+    print "                            " #to move the word blanks away from left justified
+    @answer_validation_array.each do |letter|
+      if @guessed_things.include? letter
+        print " #{letter} "
+      else
+        print " * "
+      end
+    end
+    puts "\n==================================================================\n"
+  end
+
+  def display_guessed_letters
+    if !(@guessed_things.empty?)
+      print "\nRemember that you have already guessed "
+      print "#{@guessed_things}"
+      print "."
+    end
+  end
+
+  def ask_user_for_guess
+    @last_guess = @user_guess.user_prompt
+    if @last_guess.length > 1
+      @last_guess_array = @last_guess.split(//)
+    end
+  end
+
+  def guessed_word?
+    @last_guess.length > 1
+
+  end
+
+  def add_to_guess_array
+    @guessed_things << @last_guess
+    @guessed_things.sort! #because we will display guessed letters in a reasonable fashion
+  end
+
+  def update_remaining_guesses #need to test - writing test below
+    if !(@answer_validation_array.include? @last_guess) #if the guess is not in the answer array, the user losses a guess.
+      @number_of_guesses -= 1
+      puts "Oh no!  There was no '#{ @last_guess }' in the word."
+      puts "The dinosaur is now #{ @number_of_guesses } feet away from you!!!!"
+    else
+      puts "Yipee!!! '#{ @last_guess }' was in the word."
+      puts "The dinosaur is still #{ @number_of_guesses } feet away."
+    end
+  end
+
+
+end
+
+class GuessPrompter
+
+  def initialize
+    @guessed_things = []
+  end
+
+  def user_prompt
+    @possible_letters = ('a'..'z').to_a
```

Comment:
It seems like the `@possible_letters` would not change each time the user guesses. For simplicity and consistency, it might make sense to move this instance variable into the `initialize` method, since it's something that only needs to happen once for the whole class, instead of on each prompt.